### ISSUE TITLE
TypeAdapter docs: need example of how to dump some structure to json

### ISSUE THREAD

**Issue by AntonOvsyannikov** [pydantic/pydantic#9474]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

Hello.

There is phrase `Or you may want to validate a List[SomeModel], or dump it to JSON.` in [docs](https://docs.pydantic.dev/latest/concepts/type_adapter/), but no example is provided.

I need to convert list of models to json, the only way found is to use `dump_json` method but it returns bytes for some reason.

It will be useful to get some explanation in docs about it, as well as example how to use `TypeAdapter` to dump such structure into `str` like `model_dump_json` does.

```
from pydantic import BaseModel, TypeAdapter


class Model(BaseModel):
    qq: int


a = [Model(qq=1), Model(qq=2)]

print(TypeAdapter(list[Model]).dump_json(a))
# b'[{"qq":1},{"qq":2}]'
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [X] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
@AntonOvsyannikov,

Thanks for the docs update request. This is a great first issue - we're running a sprint here at PyCon US 2024, so I've mentioned this issue to some folks who will pick it up!

**Comment by sydney-runkle** [comment]
Alright, so re the note we want to add about the bytes vs string discrepancy:

In V1, model dumping returned a str type, whereas the newly added `TypeAdapter` in V2 returns bytes. Returning bytes is more performant, but we've retained the str return for `BaseModel` for backwards compatibility reasons. 

In the `BaseModel` case, we're basically coercing bytes to strings, but bytes is most often the desired end type, hence leaving this as the return type for the `TypeAdapter` case.

**Comment by sydney-runkle** [comment]
Closed by #9477
